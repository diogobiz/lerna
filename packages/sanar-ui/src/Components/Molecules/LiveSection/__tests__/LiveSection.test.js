import React from 'react'
import { render, cleanup } from 'react-testing-library'

import ESButton from '../../../Atoms/Button'
import ESLiveSection from '../'

afterEach(cleanup)

it('renders correctly', () => {
    const { getByText, asFragment } = render(
        <ESLiveSection
            videoSrc='https://www.youtube.com/embed/j870-92BoaE'
            title='Live de Correção da prova SUS-SP 2019 - Parte 1'
            date='27/04/2019'
            name='Camila Carla Gianna'
            description={`Essa é a oportunidade de você aprender como planejar
            seus estudos em 2019! Saiba como montar um
            cronograma, quanto tempo deverá dedicar ao estudo
            por dia, quantas horas para cada matéria.`}
            action={
                <ESButton size='small' block>
                    VER LIVES ANTERIORES
                </ESButton>
            }
        />
    )

    getByText('Camila Carla Gianna')
    expect(asFragment()).toMatchSnapshot()
})
