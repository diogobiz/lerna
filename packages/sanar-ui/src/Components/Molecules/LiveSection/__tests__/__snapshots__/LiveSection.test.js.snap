// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <div
    class="ant-row-flex es-row es-live-section"
    style="margin-left: -24px; margin-right: -24px;"
  >
    <div
      class="ant-col es-col es-live-section__player ant-col-xs-24 ant-col-md-15 ant-col-lg-18"
      style="padding-left: 24px; padding-right: 24px; height: 400px;"
    >
      <div>
        <div
          class="ant-skeleton es-skeleton ant-skeleton-with-avatar ant-skeleton-active"
        >
          <div
            class="ant-skeleton-header"
          >
            <span
              class="ant-skeleton-avatar ant-skeleton-avatar-square"
              style="width: 400px; height: 400px; line-height: 400px;"
            />
          </div>
        </div>
        <iframe
          allowfullscreen=""
          frameborder="0"
          height="100%"
          src="https://www.youtube.com/embed/j870-92BoaE"
          width="100%"
        />
      </div>
    </div>
    <div
      class="ant-col es-col ant-col-xs-24 ant-col-md-9 ant-col-lg-6"
      style="padding-left: 24px; padding-right: 24px;"
    >
      <div
        class="ant-row-flex ant-row-flex-space-between es-row es-row__direction--column"
        style="height: 100%;"
      >
        <div
          class="ant-col ant-col-24 es-col mb-lg"
        >
          <div
            class="es-typography text-grey-8 mb-xs es-typography__heading es-typography__heading--h6"
            data-testid="es-typography"
          >
            Live de Correção da prova SUS-SP 2019 - Parte 1
          </div>
          <div
            class="es-typography text-grey-5 mb-md es-typography--caption"
            data-testid="es-typography"
          >
            27/04/2019
          </div>
          <div
            class="LinesEllipsis  es-typography text-grey-8 mb-md es-typography--body2"
            data-testid="es-typography"
            style="white-space: pre-wrap;"
          >
            Essa é a oportunidade de você aprender como planejar
            seus estudos em 2019! Saiba como montar um
            cronograma, quanto tempo deverá dedicar ao estudo
            por dia, quantas horas para cada matéria.
            <wbr />
          </div>
          <div
            class="es-live-section__instructor"
          >
            <span
              class="ant-avatar mr-sm ant-avatar-lg ant-avatar-circle ant-avatar-icon"
            >
              <i
                aria-label="icon: user"
                class="anticon anticon-user"
              >
                <svg
                  aria-hidden="true"
                  class=""
                  data-icon="user"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M858.5 763.6a374 374 0 0 0-80.6-119.5 375.63 375.63 0 0 0-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 0 0-80.6 119.5A371.7 371.7 0 0 0 136 901.8a8 8 0 0 0 8 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 0 0 8-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
                  />
                </svg>
              </i>
            </span>
            <div
              class="lines"
            >
              <div
                class="es-typography mb-xs es-typography--overline"
                data-testid="es-typography"
              >
                LIVE FACILITADA POR:
              </div>
              <div
                class="es-typography text-grey-8 mb-xs es-typography--strong es-typography--ellipsis"
                data-testid="es-typography"
              >
                Camila Carla Gianna
              </div>
              <div
                class="d-flex align-items-center"
              />
            </div>
          </div>
        </div>
        <div
          class="ant-col ant-col-24 es-col es-live-section__action"
        >
          <button
            class="es-button es-button__small es-button__block ant-btn"
            data-testid="es-button"
            type="button"
          >
            <span>
              VER LIVES ANTERIORES
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
